source/opt/vector_dce.cpp

```diff
@@ -351,13 +355,19 @@ void VectorDCE::AddItemToWorkListIfNeeded(
     WorkListItem work_item, VectorDCE::LiveComponentMap* live_components,
     std::vector<WorkListItem>* work_list) {
   ir::Instruction* current_inst = work_item.instruction;
-  auto it = live_components->find(current_inst->result_id());
-  if (it == live_components->end()) {
-    live_components->emplace(
-        std::make_pair(current_inst->result_id(), work_item.components));
+
+  if (current_inst->result_id() >= live_components->size()) {
+    live_components->resize(current_inst->result_id() + 100);
+  }
+
+  utils::BitVector* live_component =
```

Comment:
You could have saved the reference result from live_components->at(..) so you wouldn't do the lookup twice.... but again that's a micro-optimization that's probably not worth it.